view/adminhtml/ui_component/algolia_algoliasearch_landingpage_form.xml

```diff
@@ -0,0 +1,305 @@
+                    </item>
+                    <item name="placeholder" xsi:type="string" translate="true">product, red, woman, accessory</item>
+                    <item name="dataScope" xsi:type="string">meta_keywords</item>
+                </item>
+            </argument>
+        </field>
+        <field name="is_active">
+            <argument name="data" xsi:type="array">
+                <item name="config" xsi:type="array">
+                    <item name="formElement" xsi:type="string">checkbox</item>
+                    <item name="source" xsi:type="string">landing_page</item>
+                    <item name="default" xsi:type="number">1</item>
+                    <item name="dataType" xsi:type="string">boolean</item>
+                    <item name="prefer" xsi:type="string">toggle</item>
+                    <item name="label" xsi:type="string" translate="true">Publish Landing Page</item>
+                    <item name="notice" xsi:type="string" translate="true">If the landing page is not published, it won't be accessible on frontend.</item>
+                    <item name="dataScope" xsi:type="string">is_active</item>
+                    <item name="valueMap" xsi:type="array">
+                        <item name="true" xsi:type="number">1</item>
+                        <item name="false" xsi:type="number">0</item>
+                    </item>
+                </item>
+            </argument>
+        </field>
+        <container name="date_group">
+            <argument name="data" xsi:type="array">
+                <item name="type" xsi:type="string">group</item>
+                <item name="config" xsi:type="array">
+                    <item name="component" xsi:type="string">Magento_Ui/js/form/components/group</item>
+                    <item name="additionalClasses" xsi:type="string">admin__control-grouped-date</item>
+                    <item name="required" xsi:type="boolean">false</item>
+                    <item name="breakLine" xsi:type="boolean">false</item>
+                </item>
+            </argument>
+            <field name="date_from">
+                <argument name="data" xsi:type="array">
+                    <item name="config" xsi:type="array">
+                        <item name="additionalClasses" xsi:type="string">admin__field-date</item>
+                        <item name="source" xsi:type="string">landing_page</item>
+                        <item name="formElement" xsi:type="string">date</item>
+                        <item name="dataType" xsi:type="string">text</item>
+                        <item name="label" xsi:type="string" translate="true">Timeframe From</item>
+                        <item name="dataScope" xsi:type="string">date_from</item>
+                        <item name="notice" xsi:type="string" translate="true">If you don't set any timeframe, the landing page will be accessible indefinitely.</item>
```

Comment:
Define a timeframe when the landing page should be published (for a specific campaign for example). If no timeframe is defined, the landing page will be online unless you unpublish it manually.

* When the timeframe is over, do we unpublish the page? How does it work?

Reply:
Before displaying the page, Magento checks the timeframe, so even if the page is not marked as "unpublished", it won't be displayed if it's out of the timeframe.